### PR TITLE
[evm] report ErrInvalidCode starting London

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
as title

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] make test
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: action/protocol/execution/evm/evm.go [hunk 1]
```diff
@@ -421,74 +421,65 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
 }
 
 // evmErrToErrStatusCode returns ReceiptStatuscode which describes error type
-func evmErrToErrStatusCode(evmErr error, g genesis.Blockchain, height uint64) (errStatusCode uint64) {
+func evmErrToErrStatusCode(evmErr error, g genesis.Blockchain, height uint64) uint64 {
+	errStatusCode := iotextypes.ReceiptStatus_Failure
+	if !g.IsBering(height) {
```
Comment (Liuhaai) [inline action/protocol/execution/evm/evm.go:426]: Why is this add?

File: action/protocol/execution/evm/evm.go [hunk 2]
```diff
@@ -421,74 +421,65 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
...
-		case vm.ErrDepth:
-			errStatusCode = uint64(iotextypes.ReceiptStatus_ErrDepth)
-		case vm.ErrContractAddressCollision:
-			errStatusCode = uint64(iotextypes.ReceiptStatus_ErrContractAddressCollision)
-		case vm.ErrExecutionReverted:
-			errStatusCode = uint64(iotextypes.ReceiptStatus_ErrExecutionReverted)
-		case vm.ErrMaxCodeSizeExceeded:
-			errStatusCode = uint64(iotextypes.ReceiptStatus_ErrMaxCodeSizeExceeded)
-		case vm.ErrWriteProtection:
-			errStatusCode = uint64(iotextypes.ReceiptStatus_ErrWriteProtection)
```
Comment (dustinxie) [inline action/protocol/execution/evm/evm.go:440]: common code, same as that in `if g.IsBering(height)` below

File: action/protocol/execution/evm/evm.go (hunk 1 above)
Comment (Liuhaai) [reply action/protocol/execution/evm/evm.go:426]: what is simplified?

Comment (dustinxie) [reply action/protocol/execution/evm/evm.go:426]: see below removed block of code in `switch`

File: action/protocol/execution/evm/evm.go [hunk 3]
```diff
@@ -421,74 +421,65 @@ func executeInEVM(ctx context.Context, evmParams *Params, stateDB *StateDBAdapte
...
-			errStatusCode = uint64(iotextypes.ReceiptStatus_ErrGasUintOverflow)
-		default:
-			//This errors from go-ethereum, are not-accessible variable.
-			switch evmErr.Error() {
-			case "no compatible interpreter":
-				errStatusCode = uint64(iotextypes.ReceiptStatus_ErrNoCompatibleInterpreter)
-			default:
-				errStatusCode = uint64(iotextypes.ReceiptStatus_ErrUnknown)
-			}
+			errStatusCode = iotextypes.ReceiptStatus_ErrGasUintOverflow
```
Comment (Liuhaai) [inline action/protocol/execution/evm/evm.go:448]: These can be early returned?

**Review by Liuhaai — APPROVED** [review]

**Comment by Liuhaai** [comment]
uint64 conversion could be removed. I filed an issue for the improvement (https://github.com/iotexproject/iotex-core/issues/3240)

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**